zboxcore/sdk/deleteworker_test.go

```diff
@@ -0,0 +1,427 @@
+				allocationTx: mockAllocationTxId,
+				Consensus: Consensus{
+					consensusThresh: 50,
+					fullconsensus:   4,
+				},
+				ctx:          context.TODO(),
+				connectionID: zboxutil.NewConnectionId(),
+			}
+			objectTreeRefs := make([]fileref.RefEntity, 1)
+			refEntity, _ := req.getObjectTreeFromBlobber(blobber)
+			objectTreeRefs[0] = refEntity
+			_, err := req.getObjectTreeFromBlobber(blobber)
+			require.EqualValues(tt.wantErr, err != nil)
+			if err != nil {
+				require.EqualValues(tt.errMsg, err.Error())
+				return
+			}
+			require.NoErrorf(err, "expected no error but got %v", err)
+		})
+	}
+}
+
+func TestDeleteRequest_deleteBlobberFile(t *testing.T) {
+	const (
+		mockAllocationTxId = "mock transaction id"
+		mockClientId       = "mock client id"
+		mockClientKey      = "mock client key"
+		mockRemoteFilePath = "mock/remote/file/path"
+		mockAllocationId   = "mock allocation id"
+		mockBlobberId      = "mock blobber id"
+		mockType           = "f"
+	)
+
+	var mockClient = mocks.HttpClient{}
+	zboxutil.Client = &mockClient
+
+	client := zclient.GetClient()
+	client.Wallet = &zcncrypto.Wallet{
+		ClientID:  mockClientId,
+		ClientKey: mockClientKey,
+	}
+
+	var wg sync.WaitGroup
+
+	tests := []struct {
+		name     string
+		setup    func(t *testing.T, name string)
+		wantFunc func(require *require.Assertions, req *DeleteRequest)
+	}{
+		{
+			name: "Test_Delete_Blobber_File_Failed",
+			setup: func(t *testing.T, testName string) {
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return req.Method == "GET" &&
+						strings.HasPrefix(req.URL.Path, testName)
+				})).Run(func(args mock.Arguments) {
+					for _, c := range mockClient.ExpectedCalls {
+						c.ReturnArguments = mock.Arguments{&http.Response{
+							StatusCode: http.StatusOK,
+							Body: func() io.ReadCloser {
+								jsonFR, err := json.Marshal(fileref.ReferencePath{
+									Meta: map[string]interface{}{
+										"type": mockType,
+									},
+								})
+								require.NoError(t, err)
+								return ioutil.NopCloser(bytes.NewReader([]byte(jsonFR)))
+							}(),
+						}, nil}
+					}
+				})
+				mockClient.On("Do", mock.MatchedBy(func(req *http.Request) bool {
+					return req.Method == "DELETE" &&
+						strings.HasPrefix(req.URL.Path, testName)
+				})).Run(func(args mock.Arguments) {
+					for _, c := range mockClient.ExpectedCalls {
+						c.ReturnArguments = mock.Arguments{&http.Response{
+							StatusCode: http.StatusBadRequest,
+							Body:       ioutil.NopCloser(strings.NewReader("")),
+						}, nil}
+					}
+				})
+			},
+			wantFunc: func(require *require.Assertions, req *DeleteRequest) {
+				require.NotNil(req)
+				require.Equal(uint32(0), req.deleteMask)
+				require.Equal(float32(0), req.consensus)
+			},
+		},
+		{
+			name: "Test_Delete_Blobber_File_Success",
+			setup: func(t *testing.T, testName string) {
+				mockClient.On("Do", mock.AnythingOfType("*http.Request")).Run(func(args mock.Arguments) {
```

Comment:
Need to check `deleteBlobberFile` set `connection_id` and `path` in `http.Request`.